### ISSUE TITLE
Is it possible to indent with tabs?

### ISSUE THREAD

**Issue by r00ster91** [marktext/marktext#1592]
I would like to indent text with tabs, not spaces. If I press tab, it gives me 4 spaces. How do I use tabs?
I want to make a text like
```
Color:		red
Name:		Daniel
Location:	Street 12
```
but with tabs, so it nicely aligns.

**Comment by Jocs** [comment]
What's the difference between `tab` and `space`? it seems difficult to support `tab`, because we'll replace all `tabs` into 4 space when we parse markdown content.

**Comment by r00ster91** [comment]
A tab is `\t` (a character with the codepoint 9) and a space is ` `.

**Comment by r00ster91** [comment]
Is there any workaround to writing something like that above?

**Comment by Jocs** [comment]
>Is there any workaround to writing something like that above?

Why not use table?

**Comment by r00ster91** [comment]
That doesn't look as good as the above in my opinion. It doesn't scale well for many items and becomes pretty squeezed if you have many items.
Please make this possible. You can see this as a feature request.

**Comment by Jocs** [comment]
@r00ster91 I think `space` also can fix you problem, I still don't know why you only need `tab`, I use `space` to format you text above:

<img width="352" alt="截屏2019-11-03下午10 31 25" src="https://user-images.githubusercontent.com/9712830/68086735-d809b400-fe89-11e9-916c-b7c8b96c011d.png">

**Comment by r00ster91** [comment]
Spaces are a very ugly way to align this. Often it's not possible to align correctly. If you look at your screenshot closely you can see how inconsistent the alignment is:
![image](https://user-images.githubusercontent.com/35064754/68086845-42056780-fe50-11e9-9ac1-2d306c29e8b8.png)
If there would be tabs, there would be no problem to do things like this. It would be perfect. Spaces are very annoying to work with in this case. Especially if you now add another item that is longer than `Location`. Then you have to hammer your space bar for all the other items too to align it *somewhat* correctly (since perfect alignment isn't even possible). It's just really tedious.

**Comment by adrienbeau** [comment]
I believe Markdown is not suited for such a text-alignment feature, which is more typical of a word processor. As far as I know, no other Markdown processor implements it.

How would you configure the tab stops? How do you ensure what is correctly lined-up in one Markdown program will also be correctly lined-up in another? What looks good in HTML could be broken in PDF, with a different font, etc.

It looks like you should use a table for your example. A definition list may also work.

**Comment by fxha** [comment]
@Jocs FYI, markedjs replace all spaces with four spaces but doesn't follow markdown specification because tabs should only converted to spaces at the beginning and end of a line.

**Comment by fantazer** [comment]
i need this posible for indent code on python + pug(jade)
```pug
.block
	.block
		.block
```

but if i am copy code from app i have in my code editor.

.block
 .block
 .block

PLS HELP  =)

**Comment by Jocs** [comment]
>but if i am copy code from app i have in my code editor.

Copy content from Mark Text to other place or from other place to Mark Text?

**Comment by fantazer** [comment]
> > but if i am copy code from app i have in my code editor.
> 
> Copy content from Mark Text to other place or from other place to Mark Text?

Like this

![Alt Text](http://deploys.ru/indent.gif)

**Comment by Jocs** [comment]
@fantazer Thanks for reporting this issue, this already fixed on develop branch, and we'll release 0.16.0  in the middle of this month.

This codes bellow is copied from vscode.

<img width="863" alt="截屏2019-11-06下午11 45 43" src="https://user-images.githubusercontent.com/9712830/68313536-b447a780-00ef-11ea-8a6a-7d3055791360.png">

**Comment by brainchild0** [comment]
For code blocks, as @Jocs notes, text should be preserved as literal character sequences. (Thank you for the fix.)

For Markdown text, like other semantic languages, white-space characters are not actually characters in the text, but rather indications of word boundaries. A single space character, a single tab character, or a thousand of either is all collapsed into a single word boundary.

**Comment by leowill01** [comment]
I am still getting tab > spaces conversion inside text blocks and lists, as seen when the file is opened in or text is copied over to VS Code:

![Screen Shot 2021-03-04 at 5 25 24 PM](https://user-images.githubusercontent.com/43548130/110044360-9ce1f400-7d0e-11eb-80d0-214c88f0d5a1.png)

![Screen Shot 2021-03-04 at 5 25 42 PM](https://user-images.githubusercontent.com/43548130/110044386-a703f280-7d0e-11eb-866b-60ab137ab4d6.png)

Not only are the tabs replaced by spaces, but it doesn't even seem to follow the settings "Preferred tab width" (set to "4") or "Preferred list indentation" (set to "True tab character") in any rational manner. 

But ideally I would like to use literal tabs instead of spaces everywhere, not just code blocks.

> no other Markdown processor implements it

VS Code, RStudio, and Typora all support tabs for `.md` files. Mark Text is the first program I've used that does not. Maybe it's specifically due to the Commonmark spec because I personally use Pandoc Markdown and I've never had an issue with tabs.

**Comment by michaeljaeger95** [comment]
I agree with @leowill01, having the option to use true tabs for indentation is an essential feature for a markdown editor. 

As it stands, I'd rather stay with the last free Typora version or even nano.

**Comment by leowill01** [comment]
Sadly I though this would have been implemented by now, but it's a dealbreaker for me not being able to use tabs. It's not really about alignment, it's about indentation & character consistency. 1 indent level = 1 tab = ...1 space? 2 spaces? 4 spaces? 8 spaces? Marktext is still the only app I've ever seen that _doesn't_ give the option to use tabs.

I wanted to like Marktext because I need the autosave feature, but I've since moved to Zettlr + Panda.

**Comment by h67ma** [comment]
Any update on this? I also prefer to use tabs for indentation of lists and code. `Preferred list indentation` = `True tab character` is currently broken - lists are still being indented with two spaces instead of tab character.

**Comment by rpc-scandinavia** [comment]
Searching Preferences shows:

```
Markdown
Replace the tab with x spaces
```

Clicking that does not go to the option.
I understand here that using tabulator characters is not possible, so I will be mowing on.

**Comment by sunzx** [comment]
> What's the difference between `tab` and `space`? it seems difficult to support `tab`, because we'll replace all `tabs` into 4 space when we parse markdown content.

Replace `tab` with 4 spaces only valid at line start. 
`Tab` is more often used to align text to the next [tabulator stop](https://en.wikipedia.org/wiki/Tab_stop). in this circumstance, `tab` can be any spaces between 0 to 4.
In the example from @wooster0 , two tabs are 6 spaces in the first line, 7 spaces in the second line.

**Comment by iammihirbhagat** [comment]
I am trying to work with multi-level indentation containing both plain texts & code blocks. But it's behaving very buggy.

Overall UI/UX is good but it all goes to waste once I start writing stuff.

Mac app v0.19.1
